lib/src/address_ipv4.dart

```diff
@@ -22,6 +24,32 @@ class IPv4Address {
     return IPv4Address._internal(address);
   }
 
+  /// Creates [IPv4Address] from string [host].
+  ///
+  /// This would be useful if your Wake on LAN port has been exposed
+  /// through network that can be access from WAN.
+  ///
+  /// By default, the method will choose the first address returned
+  /// from the lookup result, use [typePredicate] to choose
+  /// which is your expected host if needed.
+  static Future<IPv4Address?> fromHost(
```

Comment:
Could we utilize a factory here instead of a static method?

Then when the `address` is null we should throw an exception/error instead of returning null. The consuming user would then handle this on their end (for example by wrapping the instantiation in a try-catch block).

Reply:
Since the `lookup` takes time, it has to be a static method.

Reply:
Ah yes, somehow missed that it's an asynchronous call. Looks good then, just as requested please remove the CHANGELOG and pubspec.yaml updates and I can approve the PR!